log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+
+    var viewUpdateFunction = (function(chart) {
+        if(this.config.tooltip.enabled){
+            this.config.tooltip.type = "arc";
+            createTooltip(div);
+            this.view = chart({el:div}).renderer(this.config.renderer).update();
+            bindTooltip(div,this.view,this.config,this.metadata);
+        } else {
+            this.view = chart({el:div}).renderer(this.config.renderer).update();
+        }
+
+        if (callbacks != null) {
+            for (var i = 0; i<callbacks.length; i++) {
+                this.view.on(callbacks[i].type, callbacks[i].callback);
+            }
+        }
+
+    }).bind(this);
+
+    if(this.config.maxLength != -1){
+        var dataset = this.spec.data[0].values;
+        var maxValue = this.config.maxLength;
+        if(dataset.length >= this.config.maxLength){
+            var allowedDataSet = [];
+            var startingPoint = dataset.length - maxValue;
+            for(var i = startingPoint; i < dataset.length;i++){
+                allowedDataSet.push(dataset[i]);
+            }
+            this.spec.data[0].values = allowedDataSet;
+        }
+    }
+
+    vg.parse.spec(this.spec, viewUpdateFunction);
+};
+
+arc.prototype.insert = function(data) {
+
+    var color = this.metadata.names[this.config.color];
+    var x = this.metadata.names[this.config.x];
+    var view =this.view;
+
+
+
+    var updated = false;
+
+    for (i = 0; i < data.length; i++) {
+        this.view.data(this.config.title)
+            .update(
+                function(d) {
+                    return d[color] == data[i][color];
+                },
+                x,
+                function(d) {
+                    updated = true;
+                    return data[i][x];
+                });
+    }
+
+    if (updated == false) {
+        view.data(this.config.title).insert(data);
+    }
+
+    this.view.update({duration: 500});
+};
+
+arc.prototype.getSpec = function() {
+    return this.spec;
+};
+
+
+function getPieMark(config, metadata){
+    var innerRadius;
+    if (config.mode == "donut") {
+        var innerRadius = config.width/5;
+    } else {
+        var innerRadius = 0;
+    }
+
+    var mark =  {
+        "type": "arc",
+        "from":  {"data": config.title},
+        "properties": {
+            "update": {
+                "x": {"field": {"group": "width"}, "mult": 0.5},
+                "y": {"field": {"group": "height"}, "mult": 0.5},
+                "startAngle": {"field": "layout_start"},
+                "endAngle": {"field": "layout_end"},
+                "innerRadius": {"value": innerRadius},
+                "outerRadius":  {"value": config.width * 0.4},
+                "fill": {"scale": "color", "field": metadata.names[config.color]},
+                "fillOpacity": {"value": 1}
+            },
+
+            "hover": {
+                "fillOpacity": {"value": 0.8}
+            }
+        }
+    };
+
+    return mark;
+};
+
+function getPieText(config, metadata){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD